hydra-node/src/Hydra/Chain/Direct.hs

```diff
@@ -156,18 +156,14 @@ withDirectChain tracer networkMagic iocp socketPath keyPair party cardanoKeys ca
                 { postTx = \tx -> do
                     traceWith tracer $ ToPost tx
 
-                    res <- timeout 10 $
-                      atomically $ do
-                        fromPostChainTx wallet (Testnet networkMagic) headState cardanoKeys tx >>= \case
-                          Nothing ->
-                            pure Nothing
-                          Just partialTx ->
-                            Just <$> finalizeTx wallet headState partialTx
-
-                    maybe
-                      (callback PostTxFailed)
-                      (atomically . writeTQueue queue)
-                      (join res)
+                    do
+                      res <-
+                        timeout 10 $
+                          atomically $
+                            fromPostChainTx wallet (Testnet networkMagic) headState cardanoKeys tx
+                              >>= finalizeTx wallet headState
+                              >>= writeTQueue queue
+                      when (isNothing res) $ throwIO (NoSeedInput @CardanoTx)
```

Comment:
I don't understand this exception here. It's not clear to me why the consequence of timing out means that there's no seed input? The "NoSeedInput" exception is already thrown within `fromPostChainTx` which seems like the right place? I recall `finalizeTx` is retrying when no payment utxo is found, so perhaps this should be the exception instead? Still, I wouldn't have it at this level, but rather, thrown by the lower levels (i.e. `finalizeTx` itself).